dao/mongo.go

```diff
@@ -214,119 +213,85 @@ func (m *MongoService) UpdateInsolvencyPractitioners(practitionersResource model
 
 	return http.StatusNoContent, nil
 }
+ 
+// GetInsolvencyPractitionerByTransactionID gets a list of all practitioners for an insolvency case with the specified transactionID
+func (m *MongoService) GetInsolvencyPractitionerByTransactionID(transactionID string) (map[string]string, string, error) {
 
-// CreatePractitionerResourceForInsolvencyCase stores an incoming practitioner to the list of practitioners for the insolvency case
-// with the specified transactionID
-func (m *MongoService) CreatePractitionerResourceForInsolvencyCase(transactionID string) (map[string]string, int, error) {
-	maxPractitioners := 5
 	var insolvencyResource models.InsolvencyResourceDto
 	var practitionersData = make(map[string]string)
 
 	collection := m.db.Collection(m.CollectionName)
+	maxPractitioners := 5
 
 	filter := bson.M{"transaction_id": transactionID}
 
 	// Retrieve insolvency case from Mongo
-	storedInsolvency := collection.FindOne(context.Background(), filter)
-	err := storedInsolvency.Err()
+	storedPractitioners := collection.FindOne(context.Background(), filter)
+	err := storedPractitioners.Err()
 	if err != nil {
 		if err == mongo.ErrNoDocuments {
 			log.Debug("no insolvency resource found for transaction id", log.Data{"transaction_id": transactionID})
-			return nil, http.StatusNotFound, fmt.Errorf("there was a problem handling your request for transaction %s not found", transactionID)
+			return nil, "", fmt.Errorf("there was a problem handling your request for transaction %s not found", transactionID)
 		}
 		log.Error(err)
-		return nil, http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
+		return nil, "", fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
 	}
 
-	err = storedInsolvency.Decode(&insolvencyResource)
+	err = storedPractitioners.Decode(&insolvencyResource)
 	if err != nil {
 		log.Error(err)
-		return nil, http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
+		return nil, "", fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
 	}
 
 	//check to ensure it is not nil from the collection
 	if len(insolvencyResource.Data.Practitioners) > 0 {
 		err = json.Unmarshal([]byte(insolvencyResource.Data.Practitioners), &practitionersData)
 		if err != nil {
 			log.Error(err)
-			return nil, http.StatusInternalServerError, fmt.Errorf("there was a problem handling json Unmarshalling %s", transactionID)
+			return nil, "", fmt.Errorf("there was a problem handling json Unmarshalling %s", transactionID)
 		}
 	}
 
 	// Check if there are already 5 practitioners in database
```

Comment:
The check for more than 5 practitioners doesn't fit well here now that it is a get not create operation - should it be moved up to the handler?

Reply:
Agreed, oversight.Done.Thanks